citesphere/src/main/java/edu/asu/diging/citesphere/web/user/concepts/types/AddConceptTypeController.java

```diff
@@ -21,22 +28,28 @@
     @Autowired
     private IUserManager userManager;
     
+    @Autowired
+    private ConceptTypeValidator conceptValidator;
+    
+    @InitBinder
+    protected void initBinder(WebDataBinder binder) {
+       binder.addValidators(conceptValidator);
+    }
+
     @RequestMapping(value="/auth/concepts/types/add", method=RequestMethod.GET)
     public String show(Model model) {
         model.addAttribute("conceptTypeForm", new ConceptTypeForm());
         return "auth/concepts/types/add";
     }
     
     @RequestMapping(value="/auth/concepts/types/add", method=RequestMethod.POST)
-    public String post(ConceptTypeForm form, Model model, Principal principal) {
-        
-        if (form.getName() != null && !form.getName().trim().isEmpty()) {
-            conceptTypeManager.create(form, userManager.findByUsername(principal.getName()));
-        } else {
+    public String post(@Validated @ModelAttribute("conceptTypeForm") ConceptTypeForm form, BindingResult result, Model model, Principal principal) {
+    	if (result.hasErrors()) {
```

Comment:
indentation

Reply:
I fixed it.
 I edited this file last month when I did not set the indentation properties.
Probably I missed it that time.